misc/keygen/src/cli.rs

```diff
@@ -0,0 +1,45 @@
+use std::ffi::OsString;
+
+use argh::FromArgs;
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// Libp2p keygen
+pub struct Cli {
+    #[argh(subcommand)]
+    subcommand: Subcommand,
+}
+
+impl Cli {
+    pub fn subcommand(&self) -> &Subcommand {
+        &self.subcommand
+    }
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand)]
+pub enum Subcommand {
+    Random(RandCmd),
+    From(FromCmd),
+}
+
+#[derive(FromArgs, PartialEq, Debug)]
+/// generate random key material
+#[argh(subcommand, name = "rand")]
+pub struct RandCmd {
+    #[argh(option, default = "String::new()")]
+    /// prefix
+    pub prefix: String,
+
+    #[argh(option, default = "String::from(\"ed25519\")")]
+    /// encryption type
+    pub r#type: String,
```

Comment:
```suggestion
    pub r#type: String,
    
    /// Path to output config file.
    pub output_path: Option<PathBuf>,
```

When generating a new key, what do you think of supporting it to be written to a config file?